### PR TITLE
WV-1931-hotfix

### PR DESCRIPTION
**Pull request by jwhurley1**
## Description

Hotfix for `release-3.21.0`.   Fixes the following issues:
- [ ] Coordinates being reversed in map marker info box and when copying to clipboard.  
- [ ] "Remove Map Marker" tooltip persists after closing/removing marker.
- [ ] Errors when adding markers in one projection, switching to another projection, then refreshing the page.

## How To Test
**Test 1**
1. Right-click to add a marker on the map. Map a note of the coordinates displayed on context menu.
2. Check that the coordinates on the info box match what is on the context menu.
3. Click the copy icon on the marker box.
4. Paste them somewhere and make sure they match the coordinates displayed on context menu.

**Test 2**
1. Add a few markers in Geographic
2. Change to Arctic/Antarctic Projection, add a marker
3. Go back to Geographic and hover (do not click) over the x to remove the markers. 
4. Click on the X.
5. The tooltip that appeared on hover should disappear along with the marker and its info box.
6. Change back to Arctic/Antarctic and repeat steps 3 through 5.

**Test 3**
1. Open Dev Tools/Console
2. Add some place markers in Geographic
3. Add some place markers in Arctic/Antarctic
4. When in a polar projection, refresh the page or save the URL and open in a new tab
5. No errors should appear on page load.

### CODE REVIEW

**Review by minniewong — COMMENTED** [review]
Another issue with the coordinates: Right click and place a marker, copy the coordinates, paste it in the "Search For Places or enter coordinates", it then places a marker with flipped coordinates.

Screen Shot 2022-03-24 at 4 14 47 PM

Placed marker with flipped coordinates:
Screen Shot 2022-03-24 at 4 15 03 PM

**Comment by jasontk19** [comment]
Same e2e test is still failing: https://github.com/nasa-gibs/worldview/runs/5682285750?check_suite_focus=true#step:8:4121

File: web/js/modules/location-search/util.js [hunk 1]
```diff
@@ -154,18 +154,20 @@ export function serializeCoordinatesWrapper(coordinates, state) {
   const { map, proj } = state;
   const serializeCoordinates = (coordinate) => {
     const coordinateValues = [coordinate.latitude, coordinate.longitude];
-    if (map.ui.selected) {
-      const coordinatesWithinExtent = areCoordinatesWithinExtent(proj, coordinateValues);
-      if (!coordinatesWithinExtent) {
-        return;
-      }
-    }
+    if (!map.ui.selected) return null;
```
Comment (jasontk19) [inline web/js/modules/location-search/util.js:157]: Not a big deal but you shouldn't need to return `null` unless whatever is calling this is looking for that value explicitly. `null`/`undefined`/`0`/`false` are all "falsy" and default `return` returns `undefined` so I would guess you're fine just returning.

File: web/js/modules/location-search/util.js [hunk 2]
```diff
@@ -154,18 +154,20 @@ export function serializeCoordinatesWrapper(coordinates, state) {
...
   };
 
-  const serializeCoordinatesArray = (coordinatesArray) => coordinatesArray.map((coordinate) => serializeCoordinates(coordinate));
+  const serializeCoordinatesArray = (coordinatesArray) => coordinatesArray
+    .map((coordinate) => serializeCoordinates(coordinate))
+    .filter((coordinate) => coordinate !== null);
   const coordinatesURL = Array.isArray(coordinates) ? serializeCoordinatesArray(coordinates) : serializeCoordinates(coordinates);
-  return coordinatesURL.join('+');
+  if (coordinatesURL.length > 0) {
+    const result = coordinatesURL.length === 1 ? coordinatesURL : coordinatesURL.join('+');
```
Comment (jasontk19) [inline web/js/modules/location-search/util.js:168]: A `.join` call on a single element array just returns that element so I think you could just:

`return coordinatesURL.join('+')`

**Comment by minniewong** [comment]
everything else looks good though!

**Comment by jwhurley1** [comment]
> Another issue with the coordinates: Right click and place a marker, copy the coordinates, paste it in the "Search For Places or enter coordinates", it then places a marker with flipped coordinates.
> 
> Screen Shot 2022-03-24 at 4 14 47 PM
> 
> Placed marker with flipped coordinates: Screen Shot 2022-03-24 at 4 15 03 PM

I've got this sorted out.  It was also adding a marker on top of the existing marker once I fixed that.  That has also been resolved.  

However, whenever there is a marker already there, it should fly to that marker.  It's not doing that currently.  Going to work on it some more, but most likely won't be until tomorrow.

**Comment by jwhurley1** [comment]
@minniewong 
> > Another issue with the coordinates: Right click and place a marker, copy the coordinates, paste it in the "Search For Places or enter coordinates", it then places a marker with flipped coordinates.
> 
> I've got this sorted out. It was also adding a marker on top of the existing marker once I fixed that. That has also been resolved.
> 
> However, whenever there is a marker already there, it should fly to that marker. It's not doing that currently. Going to work on it some more, but most likely won't be done until tomorrow.

Addressed in the latest commit.

I'm going to look into the E2E errors now.

**Comment by minniewong** [comment]
There seems to be an error with the coordinates depending on which projection you are in and which hemisphere you place the marker.

1. Change to Arctic view, add a place marker on Iceland
2. Copy the coordinates and paste into "Search for places" box
3. See error - Notification says "Provided location is outside of the map extent. Revise or try a different projection.", even though it is not. 
4. Remove marker, change to Geographic view, paste the coordinates and the map places the marker and zooms to the correct area.

Related issue:
1. In Geographic, place a place marker on Kamchatka
2. Copy the coordinates and paste into "Search for places" box
3. See error - Notification says "Provided location is outside of the map extent. Revise or try a different projection.", even though it is not. 
5. Remove marker, change to Arctic view, paste the coordinates and the map places the marker and zooms to the correct area.

**Comment by jwhurley1** [comment]
> 1. Kamchatka

Thanks!  Fixes for this incoming shortly!

**Comment by minniewong** [comment]
Looks GOOD! I have come across no more bugs

**Pull request merged**